core/loadpoint.go

```diff
@@ -252,12 +251,19 @@ func NewLoadPoint(log *util.Logger) *LoadPoint {
 
 // collectDefaults collects default values for use on disconnect
 func (lp *LoadPoint) collectDefaults() {
-	lp.onDisconnect = api.ActionConfig{
-		Mode:       lp.GetMode(),
-		MinCurrent: lp.GetMinCurrent(),
-		MaxCurrent: lp.GetMaxCurrent(),
-		MinSoC:     lp.GetMinSoC(),
-		TargetSoC:  lp.GetTargetSoC(),
+	// get reference to action config
+	actionCfg := &lp.onDisconnect
+
+	// allocate action config such that all pointer fields are fully allocated
+	if err := allocate.Zero(actionCfg); err == nil {
+		// initialize with default values
+		*actionCfg.Mode = lp.GetMode()
+		*actionCfg.MinCurrent = lp.GetMinCurrent()
+		*actionCfg.MaxCurrent = lp.GetMaxCurrent()
+		*actionCfg.MinSoC = lp.GetMinSoC()
```

Comment:
Das schliesst leere Werte mit ein- damit haben wir wieder das gleiche Problem

Reply:
Die Werte für `Mode`, `MinCurrent` und `MaxCurrent` werden in der Funktion `NewLoadPoint()` mit Default-Werten initialisiert:
https://github.com/evcc-io/evcc/blob/3b6f67e088fb7df192232b3acfc23abbbb53b46c/core/loadpoint.go#L241-L245
Wenn diese Werte in der yaml fehlen sollten, sind diese also trotzdem mit einem gültigen Wert initialisiert und nicht leer.

Fehlt `SoC.Target` wird aktuell der Wert in der Funktion NewLoadPointFromConfig() auf `100` gesetzt:
https://github.com/evcc-io/evcc/blob/3b6f67e088fb7df192232b3acfc23abbbb53b46c/core/loadpoint.go#L175-L178

Bei fehlendem `SoC.Min` in der yaml bleibt hier aktuell der Wert implizit auf 0. Was als Default-Wert für den Loadpoint ja in Ordnung geht.

Um das einheitlich zu gestalten könnte man `SoC.Target` und `Soc.Min` auch in der Funktion `NewLoadPoint()` auf die gewünschten Defaults initialisieren.

Somit sollte hier kein Problem mit leeren Werten auf der Seite des Loadpoints entstehen.